### DISCUSSION TITLE
`dependsOn` wierd behaviour -- newbie

### DISCUSSION THREAD

**Discussion by Odas0R** [vercel/turborepo#4401] [Help] [answered]
This is my local workspace `turbo.json`:

```json
{
  "extends": [
    "//"
  ],
  "pipeline": {
    "test": {
      "cache": false,
      "dependsOn": [
        "dev"
      ]
    },
    "dev": {
      "cache": false,
      "persistent": false
    }
  }
}
```

I'm using cypress to e2e, and I need to run the web server, `dev` command.

With the configuration above when I run `turbo run test` shouldn't it run both `dev` and `test` commands?

Currently only runs `dev`.

**Comment by haquezameer** [comment]
@Odas0R is your dev task forever running? perhaps that prevents `test` from ever running?

**Reply by Odas0R** [reply]
Hmmm `dev` is running a web server, so it runs forever. Shouldn't it run both tasks in parallel? `dev`, then `test` when I run `turbo run test`?

Isn't this the expected behavior when `dependsOn`?

If it isn't what's the best solution for when I have a `test` command that depends on the `dev`?

**Reply by Odas0R** [reply]
I didn't want to use the package `concurrently` but maybe this is the way? :laughing: 

Would be cool if turborepo could manage this stuff.

**Comment by mehulkar** [comment, marked as answer]
Since `test` depends on `dev`, it will never execute if `dev` doesn't exit. You can get what you want by using the `--parallel` flag, but in reality, we do not support depend on tasks that don't exit.

**Reply by Odas0R** [reply]
Yeah! Got it. Thanks :)

Since I need to run turbo run test, I'm using `concurrently` in order to achieve the above
